.github/workflows/dcm.yaml

```diff
@@ -0,0 +1,38 @@
+# A CI workflow to run the Dart Code Metrics analyzer (https://dcm.dev).
+
+# NOTE: DCM usage in DevTools is currently experimental! 
+
+# The DCM_CI_KEY and DCM_EMAIL secrets are set in the admin settings page
+# of https://github.com/flutter/devtools. These are associated with a trial
+# license of DCM, and will need to be changed if we decide to purchase an
+# actual license. The trial license will expire on April 11, 2024. 
+
+# If you need to install and run DCM locally, please refer to CONTRIBUTING.md
+# for instructions.
+
+name: Dart Code Metrics
+
+on:
+  pull_request:
```

Comment:
This will need to be changed to `pull_request_target` before submitting to run on PRs opened from forks (see https://github.com/flutter/devtools/issues/5636). However, to trigger workflows before a PR has been submitted, we need to use `pull_request` due to laxer security for `pull_request_target`: https://securitylab.github.com/research/github-actions-preventing-pwn-requests/